### PR TITLE
Improve explore() to support subfolders

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- fixes the way `scapy.contrib.scada` protocols are registered in `list_contribs()`
- adds support for subfolders handling in `explore()`:
image

The ultimate goal of `explore()` is to implement a search feature

### CODE REVIEW

**Comment by gpotter2** [comment]
i wasn't able to keep the history for Iec104: a file overwrites the others. Git fails to see the rename even when done in several commits

**Comment by guedou** [comment]
`git mv` might have worked.

File: scapy/packet.py [hunk 1]
```diff
@@ -1844,40 +1851,97 @@ def explore(layer=None):
...
-                             for x, y in _radio_values]
+            values = [(six.text_type(x), six.text_type(y))
+                      for x, y in values]
+        # Build tree
+        if action == "contribs":
+            tree = defaultdict(list)
+            for name, desc in values:
+                if "." in name:  # Folder detected
+                    parts = name.split(".")
+                    q = tree
```
Comment (guedou) [inline scapy/packet.py]: Could you change the one and two letters variables names to something more meaningful?

Comment (gpotter2) [reply scapy/packet.py]: Done. I also used `git mv` to have a commit move the file properly

Comment (guedou) [reply scapy/packet.py]: Thanks! these comments are useful =)

**Comment by guedou** [comment]
Some minor comments:
- inconsistency in layer names: some with a dot (like "Mobile IP."), other don't (like "IPsec layer")
- the cancel button should go back to the main menu instead of exiting

**Pull request merged**